### DISCUSSION TITLE
Failed to call SingleFile by Javascript.

### DISCUSSION THREAD

**Discussion by gen-root** [gildas-lormeau/SingleFile#1379] [Q&A]
Probably I met the same problem like this but I didn't understand like DesWurstes:
Call SingleFile from the JS console #820

**Describe the bug**
I added SingleFile as an extension for Chrome launched by Selenium, but I failed to call it.

**More detail**
I loaded SingleFile in Chrome launched by Selenium, and I used :
`execute_script("singlefile.saveSingleFile('{}')")` 
But I got an exception like this:
`selenium.common.exceptions.JavascriptException: Message: javascript error: singlefile is not defined`
I can see the extension in Chrome, but I can't call it in console like the link, if so I would still get this:
`Uncaught ReferenceError: singlefile is not defined 
at <anonymous>:1:38`

Where should I fix?

**Expected behavior**
Expected that Selenium could call SingleFile through `execute_script("singlefile.saveSingleFile('{}')")` , and I could even program path for each .html to save.

**Enviroment**
- **OS:Windows 10**
- **Browser:Chrome**
- **Version:121.0.6167.86**

**Comment by gildas-lormeau** [comment]
You don't need to include SingleFile as an extension. Here is an example of code showing how to inject SingleFile in the page when using Selenium: https://github.com/gildas-lormeau/single-file-cli/blob/0cb134df0638f8c7f198097d8ff830a16f8b9f9d/back-ends/webdriver-chromium.js

**Reply by wenjingYS** [reply]
Hello, the link in your reply has expired. I encountered the same problem as above. I would like a similar example for reference. Thank you very much.

**Reply by gildas-lormeau** [reply]
Thank you, I fixed the broken link in the answer. This is the last version of this file. I removed all dependencies to Selenium/Playwright/Puppeteer meanwhile in order to simplify the code and use the Chrome DevTools Protocol directly.
